packages/typespec-azure-resource-manager/src/rules/arm-resource-name-pattern.ts

```diff
@@ -0,0 +1,37 @@
+import { Program, createRule, getPattern } from "@typespec/compiler";
+
+import { getArmResources } from "../resource.js";
+
+/**
+ * Verify that a delete operation only
+ */
+export const armResourceNamePatternRule = createRule({
+  name: "arm-resource-name-pattern",
+  severity: "warning",
+  url: "https://azure.github.io/typespec-azure/docs/libraries/azure-resource-manager/rules/resource-name-pattern",
+  description: "The resource name parameter should be defined with a 'pattern' restriction.",
+  messages: {
+    default: `The resource name parameter should be defined with a 'pattern' restriction.`,
+  },
+  create(context) {
+    return {
+      root: (program: Program) => {
+        const resources = getArmResources(program);
+        for (const resource of resources) {
+          // find the name property
+          const nameProperty = [...resource.typespecType.properties.values()].find(
```

Comment:
why not do `resource.typespecType.properties.get("name")`?

Reply:
Thanks for the suggestion! Done!